lib/hooks/useSpiceSimulation.ts

```diff
@@ -0,0 +1,144 @@
+import { useState, useEffect } from "react"
+import type * as EecircuitEngine from "../types/eecircuit-engine"
+
+// Types from eecircuit-engine interface
+type RealDataType = {
+  name: string
+  type: string
+  values: number[]
+}
+type ComplexNumber = {
+  real: number
+  img: number
+}
+type ComplexDataType = {
+  name: string
+  type: string
+  values: ComplexNumber[]
+}
+type EecEngineResult =
+  | {
+      header: string
+      numVariables: number
+      variableNames: string[]
+      numPoints: number
+      dataType: "real"
+      data: RealDataType[]
+    }
+  | {
+      header: string
+      numVariables: number
+      variableNames: string[]
+      numPoints: number
+      dataType: "complex"
+      data: ComplexDataType[]
+    }
+
+export interface PlotPoint {
+  name: string // time or sweep variable
+  [key: string]: number | string
+}
+
+const parseEecEngineOutput = (
+  result: EecEngineResult,
+): { plotData: PlotPoint[]; nodes: string[] } => {
+  const columnData: Record<string, number[]> = {}
+
+  if (result.dataType === "real") {
+    result.data.forEach((col) => {
+      columnData[col.name] = col.values
+    })
+  } else if (result.dataType === "complex") {
+    result.data.forEach((col) => {
+      // For now, plot the real part of complex numbers
+      columnData[col.name] = col.values.map((v) => v.real)
+    })
+  } else {
+    throw new Error("Unsupported data type in simulation result")
+  }
+
+  const timeKey = Object.keys(columnData).find(
+    (k) => k.toLowerCase() === "time" || k.toLowerCase() === "frequency",
+  )
+  if (!timeKey) {
+    throw new Error("No time or frequency data in simulation result")
+  }
+  const timeValues = columnData[timeKey]
+  const probedVariables = Object.keys(columnData).filter((k) => k !== timeKey)
+  const plotableNodes = probedVariables.map((n) =>
+    n.replace(/v\(([^)]+)\)/i, "$1"),
+  )
+
+  const plotData: PlotPoint[] = timeValues.map((t: number, i: number) => {
+    const point: PlotPoint = { name: t.toExponential(2) }
+    probedVariables.forEach((variable, j) => {
+      point[plotableNodes[j]] = columnData[variable][i]
+    })
+    return point
+  })
+
+  return { plotData, nodes: plotableNodes }
+}
+
+type WorkerMessage =
+  | {
+      type: "result"
+      result: EecEngineResult
+    }
+  | { type: "error"; error: string }
+
+export const useSpiceSimulation = (spiceString: string | null) => {
+  const [plotData, setPlotData] = useState<PlotPoint[]>([])
+  const [nodes, setNodes] = useState<string[]>([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState<string | null>(null)
+
+  useEffect(() => {
+    if (!spiceString) {
+      setIsLoading(false)
+      setPlotData([])
+      setNodes([])
+      setError(null)
+      return
+    }
+    setIsLoading(true)
+    setError(null)
+    setPlotData([])
+    setNodes([])
+
+    const worker = new Worker(
+      new URL("../workers/spice-simulation.worker.ts", import.meta.url),
```

Comment:
this will only work in `bun`, you have to bundle the worker code (same as eval) then import the bundle.

Reply:
yea i think the vite server does some crazy stuff but it's loaded dynamically and that won't work when it's used as a module, you have to bundle the webworker code (without bundling the eecircuit engine)

<img width="3182" height="500" alt="image" src="https://github.com/user-attachments/assets/98bbdaf8-12df-43ac-9df1-2a8d22938118" />